File: sdk-python/examples/prices/auto_paginate.py
"""Iterate every active price across all products, transparently fetching each
page as the previous one drains.

Run with: python examples/prices/auto_paginate.py
"""

from __future__ import annotations

from threecommon import ThreeCommon
from threecommon.prices import ListParams


def main() -> None:
    count = 0
    recurring = 0
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        for price in client.prices.list_auto_paginate(ListParams(active=True)):
            count += 1
            if price.type == "recurring":
                recurring += 1

    print(f"iterated {count} active prices")
    print(f"{recurring} are recurring")


if __name__ == "__main__":
    main()

File: sdk-node/examples/prices/auto-paginate.ts
/**
 * Iterate every active price across all products, transparently fetching each
 * page as the previous one drains.
 *
 * Run:
 *   npx tsx examples/prices/auto-paginate.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

let count = 0
let recurring = 0
for await (const price of client.prices.listAutoPaginate({ active: true })) {
  count += 1
  if (price.type === 'recurring') recurring += 1
}

console.log(`iterated ${String(count)} active prices`)
console.log(`${String(recurring)} are recurring`)
